api/envoy/config/listener/v3/listener_components.proto

```diff
@@ -65,6 +65,18 @@ message Filter {
 // ``www.example.com``, then ``*.example.com``, then ``*.com``, then any filter
 // chain without ``server_names`` requirements).
 //
+// Yet another approach to reason about the filter chain matches.
+// Suppose you have N filter chains in hand. Sieve the filter chain set in the above 8 rounds.
+// In each round, filter chains matching the attributes best survives to the next round.
+// The listener guarantees at most 1 filter chain left after all the rounds.
```

Comment:
What does "best matching" mean here? Can there be more than one best match? I think it's important to be very precise here, because we're trying to reduce confusion, and any ambiguity increases it.